pkg/controller/accountpool/accountpool_controller.go

```diff
@@ -114,7 +114,7 @@ func (r *ReconcileAccountPool) Reconcile(request reconcile.Request) (reconcile.R
 	for _, account := range accountList.Items {
 		// We don't want to count reused accounts here, filter by LegalEntity.ID
 		if !account.Status.Claimed && account.Spec.LegalEntity.ID == "" {
-			if account.Status.State != "Failed" {
+			if account.Status.State == string(awsv1alpha1.AccountCreating) || account.Status.State == string(awsv1alpha1.AccountPending) || account.Status.State == string(awsv1alpha1.AccountPendingVerification) || account.Status.State == string(awsv1alpha1.AccountReady) {
```

Comment:
This should be put into a function to check because all of these checks are hard to encapsulate what logic we care about.

Reply:
Or, even better, to put this into the Account object to ask something like `if account.IsFailed()` or similar so that the `Account` object is responsible for knowing how to answer that question; the controller here doesn't need to know what it means for an `Account` to be in a good state, just that it can get an answer.